dashboard/test/models/regional_partner_test.rb

```diff
@@ -65,4 +65,59 @@ class RegionalPartnerTest < ActiveSupport::TestCase
       regional_partner.program_manager = program_manager.id
     end
   end
+
+  test 'find_by_region finds matching regional_partner' do
+    # partners and their mappings
+
+    # regional_partner_WA
+    # regional_partner_WA_98105
+    # regional_partner_98105
+    # regional_partner_NY
+    regional_partner_ny = create :regional_partner, name: "partner_NY"
+    regional_partner_ny.mappings.find_or_create_by(state: "NY")
+
+    # regional_partner_MA_02138
+    regional_partner_ma_02138 = create :regional_partner, name: "partner_MA_02138"
+    regional_partner_ma_02138.mappings.find_or_create_by(state: "MA")
+    regional_partner_ma_02138.mappings.find_or_create_by(zip_code: "02138")
+
+    # regional_partner_CA_94305
+    regional_partner_ca_94305 = create :regional_partner, name: "partner_CA_94305"
+    regional_partner_ca_94305.mappings.find_or_create_by(state: "CA")
+    regional_partner_ca_94305.mappings.find_or_create_by(zip_code: "94305")
+
+    # regional_partner_70808
+    regional_partner_70808 = create :regional_partner, name: "partner_70808"
+    regional_partner_70808.mappings.find_or_create_by(zip_code: "70808")
+
+    # regional_partner_no_mappings
+
+    # queries
+
+    # state=nil/zip=nil [return nil (no match)]
+    assert_nil RegionalPartner.find_by_region(nil, nil)
+
+    # state=NY [search by state only where there is only one match]
+    assert_equal RegionalPartner.find_by_region(nil, "NY"), regional_partner_ny
```

Comment:
Reverse the args. [assert_equal](http://ruby-doc.org/stdlib-2.1.0/libdoc/test/unit/rdoc/Test/Unit/Assertions.html#method-i-assert_equal) takes expected followed by actual.

Same below.